File: android/app/src/main/kotlin/run/nuri/getagrip/ui/components/BatteryDisplay.kt
// SPDX-License-Identifier: MPL-2.0
// Original contributions Copyright 2026 Nuri Bruner.

package run.nuri.getagrip.ui.components

import kotlin.math.roundToInt

object BatteryDisplay {
    fun percentage(fraction: Double): Int =
        if (fraction.isFinite()) (fraction.coerceIn(0.0, 1.0) * 100).roundToInt() else 0
}

File: Sources/UI/Components/BatteryDisplay.swift
// SPDX-License-Identifier: MPL-2.0
// Original contributions Copyright 2026 Nuri Bruner.

enum BatteryDisplay {
    static func percentage(_ fraction: Double) -> Int {
        guard fraction.isFinite else { return 0 }
        return Int((min(max(fraction, 0), 1) * 100).rounded())
    }
}
